python/dgl/distributed/kvstore.py

```diff
@@ -792,6 +792,7 @@ def push_handler(data_store, name, local_offset, data)
         func : callable
             The function to be called.
         """
+        self.barrier()
```

Comment:
why do we need a barrier here?

Reply:
We need this barrier. Otherwise, we could change the push_handler before the previous push message finishing its job.